lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -3225,6 +3225,14 @@ final public function getDiscriminatorColumn(): array
      */
     public function setDiscriminatorMap(array $map)
     {
+        if (count(array_flip($map)) !== count($map)) {
+            Deprecation::trigger(
+                'doctrine/orm',
+                'https://github.com/doctrine/orm/issues/3519',
+                'Mapping a class to multiple discriminator values is deprecated.'
```

Comment:
Let's add the class that has multiple mappings and which those are with `$this->name` and `array_keys(array_filter(array_count_values($map), function ($value) { return $value > 1; }))`?